packages/js/src/dashboard/widgets/search-ranking-compare-widget.js

```diff
@@ -0,0 +1,196 @@
+				<SkeletonLoader className="yst-text-center yst-text-sm yst-mt-2">Dummy</SkeletonLoader>
+				<SkeletonLoader className="yst-text-center yst-text-sm yst-mt-2 yst-font-semibold">- 13%</SkeletonLoader>
+			</div>
+			<div className="yst-h-px yst-w-full yst-bg-slate-200 yst-my-6 @6xl:yst-h-20 @6xl:yst-w-px @6xl:yst-my-1" />
+
+			<div className="yst-flex yst-flex-col yst-relative yst-items-center yst-w-72 yst-content-around">
+				<div className="yst-absolute yst-end-6 yst-top-2">
+					<InfoTooltip>
+						<TooltipContent
+							tooltipLocalizedString={ __( "Average position is the average position of your site in search results over the last 28 days.", "wordpress-seo" ) }
+						/>
+					</InfoTooltip>
+				</div>
+				<SkeletonLoader className="yst-text-center yst-text-2xl yst-font-bold yst-text-slate-900">12345</SkeletonLoader>
+				<SkeletonLoader className="yst-text-center yst-text-sm yst-mt-2">Dummy</SkeletonLoader>
+				<SkeletonLoader className="yst-text-center yst-text-sm yst-mt-2 yst-font-semibold">- 13%</SkeletonLoader>
+			</div>
+		</div>
+	);
+};
+
+/* eslint-disable complexity */
+/**
+ * @param {TimeBasedData[]} data The data.
+ * @returns {SearchRankingCompareData} The transformed data.
+ */
+const transformData = ( data ) => {
+	if ( Object.keys( data[ 0 ].current ).length === 2 ) {
+		return {
+			impressions: {
+				value: data[ 0 ].current.total_impressions || NaN,
+				delta: getDifference( data[ 0 ]?.current?.total_impressions || NaN, data[ 0 ]?.previous?.total_impressions || NaN ),
+			},
+			clicks: {
+				value: data[ 0 ].current.total_clicks || NaN,
+				delta: getDifference( data[ 0 ]?.current?.total_clicks || NaN, data[ 0 ].previous.total_clicks || NaN ),
+			},
+		};
+	}
+
+	return {
+		impressions: {
+			value: data[ 0 ].current.total_impressions || NaN,
+			delta: getDifference( data[ 0 ].current.total_impressions || NaN, data[ 0 ].previous.total_impressions || NaN ),
+		},
+		clicks: {
+			value: data[ 0 ].current.total_clicks || NaN,
+			delta: getDifference( data[ 0 ].current.total_clicks || NaN, data[ 0 ].previous.total_clicks || NaN ),
+		},
+		ctr: {
+			value: data[ 0 ].current.average_ctr || NaN,
+			delta: getDifference( data[ 0 ].current.average_ctr || NaN, data[ 0 ].previous.average_ctr || NaN ),
+		},
+		position: {
+			value: data[ 0 ].current.average_position || NaN,
+			delta: getDifference( data[ 0 ].current.average_position || NaN, data[ 0 ].previous.average_position || NaN ),
+		},
+	};
+};
+
+/**
+ * @param {function} dataTransformer The data transformer.
+ * @param {import("../services/comparison-metrics-data-formatter")} dataFormatter The data formatter.
+ * @returns {function(?TimeBasedData[]): SearchRankingCompareData} Function to format the widget data.
+ */
+export const createIcsaDataFormatter = ( dataTransformer, dataFormatter ) => ( initData ) => {
+	const data = dataTransformer( initData );
```

Comment:
Why are you combining the transformer and the formatter instead of chaining them? That way the formatter here can stay just formatting.